misc/scripts/install-local.sh

```diff
@@ -905,52 +972,335 @@ if ! is_package_installed "${name}"; then
+        cd ..
+        gather_down
+    else
+        export pkgbase="${PWD}"
+    fi
+}
+
+function net_down() {
+    fancy_message info "Downloading ${BPurple}${dest}${NC}"
+    download "$url" "$dest" || fail_down
+}
+
+function hashcheck_down() {
+    if [[ -n ${expectedHash} ]]; then
+        fancy_message sub "Checking hash ${YELLOW}${expectedHash:0:8}${NC}[${YELLOW}...${NC}]"
+        hashcheck "${dest}" "${expectedHash}" || return 1
+    fi
+}
+
+function genextr_down() {
+    hashcheck_down
+    local extract=true
+    for keep_archive in "${noextract[@]}"; do
+        if [[ ${keep_archive} == "${dest}" ]]; then
+            extract=false
+            break
+        fi
+    done
+    if ${extract}; then
+        fancy_message sub "Extracting ${CYAN}${dest}${NC}"
+        ${ext_method} "${dest}" 1>&1 2> /dev/null
+        if [[ -f ${dest} ]]; then
+            rm -f "${dest}"
+        fi
+    fi
+    if [[ ${source[i]} == "${source[0]}" && ${extract} == "true" ]]; then
+        cd ./*/ 2> /dev/null || {
+            error_log 1 "install $PACKAGE"
+            fancy_message warn "Could not enter into the downloaded archive"
+        }
+        export pkgbase="${PWD}"
+    else
+        gather_down
+    fi
+}
+
+function deb_down() {
+    hashcheck_down
+    if type -t pre_install &> /dev/null; then
+        if ! pre_install; then
+            error_log 5 "pre_install hook"
+            fancy_message error "Could not run preinst hook successfully"
+            exit 1
+        fi
+    fi
+    if sudo apt install -y -f ./"${dest}" 2> /dev/null; then
+        log
+        if [[ -f /tmp/pacstall-pacdeps-"$name" ]]; then
+            sudo apt-mark auto "${gives:-$name}" 2> /dev/null
+        fi
+        if type -t post_install &> /dev/null; then
+            if ! post_install; then
+                error_log 5 "post_install hook"
+                fancy_message error "Could not run post_install hook successfully"
+                exit 1
+            fi
+        fi
+        fancy_message info "Storing pacscript"
+        sudo mkdir -p "/var/cache/pacstall/$PACKAGE/${full_version}"
+        if ! cd "$DIR" 2> /dev/null; then
+            error_log 1 "install $PACKAGE"
+            fancy_message error "Could not enter into ${DIR}"
+            exit 1
+        fi
+        sudo cp -r "${pacfile}" "/var/cache/pacstall/$PACKAGE/${full_version}"
+        sudo chmod o+r "/var/cache/pacstall/$PACKAGE/${full_version}/$PACKAGE.pacscript"
         fancy_message info "Cleaning up"
         cleanup
-        return 1
+        exit 0
+    else
+        fancy_message error "Failed to install the package"
+        error_log 14 "install $PACKAGE"
+        sudo apt purge "${gives:-$name}" -y > /dev/null
+        clean_fail_down
+    fi
+}
+
+function file_down() {
+    fancy_message info "Copying local archive ${BPurple}${dest}${NC}"
+    cp -r "${url}" "${dest}" || fail_down
+    genextr_declare
+    if [[ -n ${ext_method} ]]; then
+        genextr_down
+    fi
```

Comment:
I need to correct this so that elif [[ source[i] == source[0] && -d $dest ]]; then cd $dest && export pkgbase=$PWD; fi 

and then maybe end it with a gather_down, I have to think if that's necessary